src/fdcache_page.cpp

```diff
@@ -845,7 +845,10 @@ bool PageList::Serialize(CacheFileStat& file, ino_t inode)
         S3FS_PRN_ERR("failed to write stats(%d)", errno);
         return false;
     }
-
+    if(0 != fsync(file.GetFd())){
```

Comment:
I don't understand how this helps.  `fsync` should take the data from the kernel buffers and write it to disk.  Without `fsync` another process should still be able to write from the kernel buffers.  If instead the data were in the user buffers, e.g., `fwrite`, and the code did not call `fflush`, then another process could not read the data.  But since this calls `pwrite` it should not be a problem.

Reply:
You're right, fsync doesn't make sense.
I'll continue investigating other causes, and close this PR eventually (without merging).